observer/holder/nodesHolder_test.go

```diff
@@ -0,0 +1,284 @@
+package holder
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-proxy-go/data"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNodesHolder_GetLastSyncedNodes(t *testing.T) {
```

Comment:
perhaps an extra concurrency test?

example:
```
func TestNodesHolder_Concurrency(t *testing.T) {
	t.Parallel()

	syncedNodes := []*data.NodeData{{Address: "addr0", ShardId: core.MetachainShardId}, {Address: "addr1", ShardId: 0}}
	fallbackNodes := []*data.NodeData{{Address: "fallback-addr0", ShardId: core.MetachainShardId, IsFallback: true}, {Address: "fallback-addr1", ShardId: 0, IsFallback: true}}
	allNodes := append(syncedNodes, fallbackNodes...)
	shardIds := []uint32{0, core.MetachainShardId}

	nh, err := NewNodesHolder(syncedNodes, fallbackNodes, shardIds, data.AvailabilityAll)
	require.NoError(t, err)

	numCalls := 1000
	wg := sync.WaitGroup{}
	wg.Add(numCalls)

	for i := 0; i < numCalls; i++ {
		go func(idx int) {
			switch idx {
			case 0:
				nh.GetSyncedNodes()
			case 1:
				nh.GetSyncedFallbackNodes()
			case 2:
				nh.GetOutOfSyncNodes()
			case 3:
				nh.GetOutOfSyncFallbackNodes()
			case 4:
				nh.GetLastSyncedNodes()
			case 5:
				nh.UpdateNodes(allNodes)
			}
			wg.Done()
		}(i % 6)
	}

	wg.Wait()
}
```

Reply:
added a concurrency test for the refactored version